src/main/java/io/github/cottonmc/energy/api/EnergyAttributeProvider.java

```diff
@@ -0,0 +1,11 @@
+package io.github.cottonmc.energy.api;
+
+import net.minecraft.nbt.CompoundTag;
+
+public interface EnergyAttributeProvider {
+    default EnergyAttribute getEnergy(CompoundTag tag) {
+        return fromTag(tag);
+    }
+    EnergyAttribute fromTag(CompoundTag tag);
```

Comment:
This could conflict with a vanilla `void fromTag(CompoundTag)` if implemented on some type that has that.

Reply:
It shouldn’t. It would just require it.

Reply:
I think it might cause an issue, yeah. Method signatures don't include the return type.

Reply:
Ohh, right. But I’m away from my PC by now again. I will solve it tomorrow by my time, or you may if you want to